Packages/com.unity.inputsystem/InputSystem/Editor/AssetEditor/InputActionTreeViewItems.cs

```diff
@@ -0,0 +1,530 @@
+        }
+
+        public override bool AcceptsDrop(ActionTreeItemBase item)
+        {
+            return item is PartOfCompositeBindingTreeItem;
+        }
+
+        public override bool GetDropLocation(Type itemType, int? childIndex, ref SerializedProperty array, ref int arrayIndex)
+        {
+            // Drop part binding into composite.
+            if (itemType == typeof(PartOfCompositeBindingTreeItem))
+            {
+                array = arrayProperty;
+
+                // Adjust child index by index of composite item itself.
+                arrayIndex = childIndex != null
+                    ? this.arrayIndex + childIndex.Value
+                    : this.arrayIndex + 1 + InputActionSerializationHelpers.GetCompositePartCount(array, this.arrayIndex);
+
+                return true;
+            }
+
+            // Drop other bindings next to us.
+            if (typeof(BindingTreeItem).IsAssignableFrom(itemType))
+            {
+                array = arrayProperty;
+                arrayIndex = this.arrayIndex + 1 +
+                    InputActionSerializationHelpers.GetCompositePartCount(array, this.arrayIndex);
+                return true;
+            }
+
+            return false;
+        }
+
+        public static CompositeBindingTreeItem AddTo(TreeViewItem parent, SerializedProperty bindingProperty)
+        {
+            var item = new CompositeBindingTreeItem(bindingProperty);
+
+            item.depth = parent.depth + 1;
+            item.displayName = !string.IsNullOrEmpty(item.name) ? item.name : ObjectNames.NicifyVariableName(item.path);
+            parent.AddChild(item);
+
+            return item;
+        }
+    }
+
+    /// <summary>
+    /// Tree view item for bindings that are parts of composites.
+    /// </summary>
+    /// <see cref="InputBinding.isPartOfComposite"/>
+    internal class PartOfCompositeBindingTreeItem : BindingTreeItem
+    {
+        public PartOfCompositeBindingTreeItem(SerializedProperty bindingProperty)
+            : base(bindingProperty)
+        {
+        }
+
+        public override GUIStyle colorTagStyle => Styles.pinkRect;
+        public override bool canRename => false;
+
+        public override string expectedControlLayout
+        {
+            get
+            {
+                if (m_ExpectedControlLayout == null)
+                {
+                    var partName = name;
+                    var compositeName = ((CompositeBindingTreeItem)parent).name;
+                    var layoutName = InputBindingComposite.GetExpectedControlLayoutName(compositeName, partName);
+                    m_ExpectedControlLayout = layoutName ?? "";
+                }
+
+                return m_ExpectedControlLayout;
+            }
+        }
+
+        private string m_ExpectedControlLayout;
+
+        public static PartOfCompositeBindingTreeItem AddTo(TreeViewItem parent, SerializedProperty bindingProperty)
```

Comment:
new warning   warning CS0108: 'PartOfCompositeBindingTreeItem.AddTo(TreeViewItem, SerializedProperty)' hides inherited member 'BindingTreeItem.AddTo(TreeViewItem, SerializedProperty)'. Use the new keyword if hiding was intended.